pemi/pipe.py

```diff
@@ -172,6 +172,31 @@ def __init__(self):
         self.connections.append(conn)
         return conn
 
+    def analyze(self, plumbing=None, cur_name='', ret_arr={}):  # pylint disable=W0102
```

Comment:
`analyze` seems a little too generic.  How about `summarize`?

Also, when disabling pylint messages, use the human-readable string instead of the code (in this case I think it's `dangerous-default`?).

I don't like disabling dangerous-default.  I've certainly introduced bugs by ignoring this.  As an alternative, set `ret_arr=None` in the arguments, then add a line like `ret_arr = ret_arr or {}`.

Reply:
Lastly, can you write a simple pytest test for this new function?